packages/ui/src/components/CognitoAuth/hooks/useSigv4Client/utils/getCredentials/index.ts

```diff
@@ -0,0 +1,55 @@
+/** *******************************************************************************************************************
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+  
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+  
+      http://www.apache.org/licenses/LICENSE-2.0
+  
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.                                                                              *
+ ******************************************************************************************************************** */
+import { AwsCredentialIdentity } from '@aws-sdk/types';
+import { CognitoUserSession, CognitoUser } from 'amazon-cognito-identity-js';
+import { CognitoIdentityClient } from '@aws-sdk/client-cognito-identity';
+import { fromCognitoIdentityPool } from '@aws-sdk/credential-provider-cognito-identity';
+
+const getCredentials = (cognitoUser: CognitoUser, region?: string, identityPoolId?: string, userPoolId?: string) => {
+    return new Promise<AwsCredentialIdentity>(async (resolve, reject) => {
+        if (!region) {
+            reject(new Error('region is empty'));
```

Comment:
IDK, but maybe it will be better to create a new type error:
```js
class EmptyFieldError extends Error {
  constructor(message) {
    super(message);
    this.name = 'EmptyFieldError';
  }
}

...
reject(new EmptyFieldError('region is empty'));
```